### ISSUE TITLE
Question about connecting to a stream

### ISSUE THREAD

**Issue by scottforsyth** [tempo-riz/deepgram_speech_to_text#8]
Hello Thibault,

I successfully got your Deepgram package to send and process a .wav file. It works great. It's very easy and useful.

However, I'm trying to work with a stream but haven't had any luck. I may be missing a simple concept. I'm quite new to Flutter/Dart.

I'm using version 1.0.5 due to dependency requirements on my current project, so I can't upgrade to the latest. 

I tried with my own stream. I insert frames into that stream.

Then, my code is pretty simple.

At the class level:

```
  late Deepgram deepgram;
  late DeepgramLiveTranscriber transcriber;
  StreamController<List<int>> _deepgramAudioStreamController = StreamController<List<int>>();
  Map<String, dynamic> params = {
    'model': 'nova-2-general', // or conversationalai
    'detect_language': false,
    'language': 'en-US',
    'smart-format': true,
    'encoding': 'linear16',
    'filler_words': false,
    'punctuation': true,
    'utterence_end_ms': 1000,
  };
```

Within init:
`deepgram = Deepgram(deepgramApiKey, baseQueryParams: params);`

Within a different method, I'm filling _deepgramAudioStreamController.stream with the stream frames. I believe that's working well since I have it working for a couple of other things that do work.

And, within a _startDeepgramListening() method:
  ```
transcriber = deepgram.createLiveTranscriber(_deepgramAudioStreamController.stream);
  transcriber.start();
  transcriber.jsonStream.listen((json) {
    print(json);
  });
```

And, in a separate method called _processRecordedAudio():
`  transcriber.close();`

Should I expect that a breakpoint on the print(json) line would catch? It doesn't catch and it doesn't print anything. No errors are sent either though. Note that sending a wav file does work.

I also tried a simple option with all the code in the _startDeepgramListening() method:
```
  Stream<String> jsonStream = deepgram.transcribeFromLiveAudioStream(_deepgramAudioStreamController.stream);
  transcriber.start();
    transcriber.jsonStream.listen((json) {
      print(json);
    });
```

However, nothing prints.

Am I missing something dumb? I'm not sure where in the Deepgram docs I can find more info since all the examples seem to use their components (JavaScript, C#, Go, etc.).

I see in the Deepgram dashboard that the count of requests increases for each attempt. I don't know how to wire up the callback.

Thanks!
Scott

**Comment by tempo-riz** [comment]
Hey @scottforsyth, the code samples you showed seem fine. Are you sure the data you put in your input stream is legit? For example, does it have the same encoding and rate that you specified in the options?

I made a Flutter example that you can start from: [Flutter Example](https://github.com/tempo-riz/deepgram_speech_to_text/tree/main/example/flutter_example).

I’m also doing something similar in the package's test, so you can take a look here: [Deepgram Test](https://github.com/tempo-riz/deepgram_speech_to_text/blob/main/test/deepgram_test.dart).

What I did there: read an audio file once, then create a stream where I put that data.

Hope it helps!

**Comment by scottforsyth** [comment]
Hey Thibault,

You're absolutely right. I used your example and it worked right away. 

As you suggested, I did have my custom stream wrong. I needed to convert it to pcm16bits. I thought that it was, but it wasn't. After updating that (my code below), it works beautifully with my custom stream.

```
  Uint8List _convertToPcmS16LE(List<int> frame) {
    ByteData byteData = ByteData(frame.length * 2);
    for (int i = 0; i < frame.length; i++) {
      byteData.setInt16(i * 2, frame[i], Endian.little);
    }
    return byteData.buffer.asUint8List();
  }
```

Thanks for your quick reply and pointing me in the right direction. And thanks again for providing the Deepgram package!
